libraries/freertos_plus/standard/https/src/private/iot_https_internal.h

```diff
@@ -448,6 +452,9 @@ typedef struct _httpsRequest
     struct _httpsResponse *pHttpsResponse;     /**< @brief Response associated with request. This is initialized during IotHttpsClient_InitializeRequest(), then returned to the application in IotHttpsClient_SendAsync() and IotHttpsClient_SendSync(). */
     struct _httpsConnection *pHttpsConnection;   /**< @brief Connection associated with request. This is set during IotHttpsClient_SendAsync(). It is needed for the asynchronous workflow to use to send data given the reqHandle only in the callback. */
     bool isNonPersistent;   /**< @brief Non-persistent flag to indicate closing the connection immediately after receiving the response. */
+    bool isAsync;           /**< @brief This is set to true if this request is to be sent asynchronously. Set to false otherwise. */
+    void * pUserPrivData;   /**< @brief User private data to hand back in the asynchronous callbacks for context. */
+    IotHttpsClientCallbacks_t* pCallbacks;   /**< @brief Pointer to the asynchronous request callbacks. */
```

Comment:
can we make this cleaner by collecting all data structures for async into a async structure ? This will become difficult to maintain if we dont organize it. 

This can be potentially done quickly after GA too.

Reply:
Let's do this after GA. I would need to find all usages of each member and edit them to go through the async structure. I am not sure if it is useful to put just two items in the structure. All other items are shared with both processes.

Reply:
yes, agreed. Something for post GA